fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonElectionSerializer.java

```diff
@@ -0,0 +1,152 @@
+
+    // Deserialize the enums
+    ElectionVersion electionVersion =
+        ElectionVersion.valueOf(jsonObject.get("electionVersion").getAsString());
+    String stateString = jsonObject.get("state").getAsString();
+    EventState state = stateString.isEmpty() ? null : EventState.valueOf(stateString);
+
+    // Deserialize collections
+    JsonArray electionQuestionsJsonArray = jsonObject.get("electionQuestions").getAsJsonArray();
+    List<ElectionQuestion> electionQuestions = new ArrayList<>();
+    for (JsonElement electionQuestionJsonElement : electionQuestionsJsonArray) {
+      electionQuestions.add(
+          context.deserialize(electionQuestionJsonElement, ElectionQuestion.class));
+    }
+
+    Map<PublicKey, List<Vote>> votesBySender = new HashMap<>();
+    JsonObject votesBySenderObject = jsonObject.get("votesBySender").getAsJsonObject();
+    for (Map.Entry<String, JsonElement> entry : votesBySenderObject.entrySet()) {
+      PublicKey senderPublicKey = new PublicKey(entry.getKey());
+      JsonArray votesBySenderJsonArray = entry.getValue().getAsJsonArray();
+      List<Vote> votesBySenderPk = new ArrayList<>();
+      for (JsonElement element : votesBySenderJsonArray) {
+        votesBySenderPk.add(context.deserialize(element, Vote.class));
+      }
+      votesBySender.put(senderPublicKey, votesBySenderPk);
+    }
+
+    Map<PublicKey, MessageID> messageMap = new HashMap<>();
+    JsonObject messageMapObject = jsonObject.get("messageMap").getAsJsonObject();
+    for (Map.Entry<String, JsonElement> entry : messageMapObject.entrySet()) {
+      PublicKey senderPublicKey = new PublicKey(entry.getKey());
+      MessageID messageId = new MessageID(entry.getValue().getAsString());
+      messageMap.put(senderPublicKey, messageId);
+    }
+
+    Map<String, Set<QuestionResult>> results = new HashMap<>();
+    JsonObject resultsObject = jsonObject.get("results").getAsJsonObject();
+    for (Map.Entry<String, JsonElement> entry : resultsObject.entrySet()) {
+      String questionId = entry.getKey();
+      JsonArray resultsJsonArray = entry.getValue().getAsJsonArray();
+      Set<QuestionResult> resultsByQuestion = new HashSet<>();
+      for (JsonElement element : resultsJsonArray) {
+        resultsByQuestion.add(context.deserialize(element, QuestionResult.class));
+      }
+      results.put(questionId, resultsByQuestion);
+    }
+
+    return new Election(
+        id,
+        name,
+        creation,
+        channel,
+        start,
+        end,
+        electionQuestions,
+        electionKey,
+        electionVersion,
+        votesBySender,
+        messageMap,
+        state,
+        results);
+  }
+
+  @Override
+  public JsonElement serialize(
+      Election election, Type typeOfSrc, JsonSerializationContext context) {
+    JsonObject jsonObject = new JsonObject();
+
+    jsonObject.add("channel", context.serialize(election.getChannel(), Channel.class));
+
```

Comment:
Like in `deserialize`, you could add a comment about what is being serialized in each code block, it makes the code easier to read.